zap/src/parser/convert.rs

```diff
@@ -166,228 +204,326 @@ impl<'src> SyntaxEvDecl<'src> {
 				}
 
-				Enum::Unit(enumerators.into_iter().map(|v| v.into_config()).collect())
+				Enum::Unit(enumerators.iter().map(|e| e.name).collect())
 			}
 
 			SyntaxEnumKind::Tagged { tag, variants } => {
-				if variants.is_empty() {
-					state.push_report(Report::AnalyzeEmptyEnum { span })
-				}
-
-				let tag_name = tag.into_config();
+				let tag_name = tag.value;
+
+				let variants = variants
+					.iter()
+					.map(|variant| {
+						if variant.1.fields.iter().any(|field| field.0.name == tag_name) {
+							self.report(Report::AnalyzeEnumTagUsed {
+								tag_span: tag.span(),
+								used_span: variant.0.span(),
+								tag: tag_name,
+							});
+						}
+
+						(variant.0.name, self.struct_ty(&variant.1))
+					})
+					.collect();
 
 				Enum::Tagged {
 					tag: tag_name,
-					variants: variants
-						.into_iter()
-						.map(|(name, value)| {
-							if let Some(field) = value.fields.iter().find(|field| field.0.name == tag_name) {
-								state.push_report(Report::AnalyzeEnumTagUsed {
-									tag_span: tag.span(),
-									used_span: field.0.span(),
-									tag: tag_name,
-								});
-							}
-
-							(name.into_config(), value.into_config(state))
-						})
-						.collect(),
+					variants,
 				}
 			}
 		}
 	}
-}
 
-impl<'src> SyntaxStruct<'src> {
-	fn into_config(self, state: &mut ConvertState<'src>) -> Struct<'src> {
+	fn struct_ty(&mut self, ty: &SyntaxStruct<'src>) -> Struct<'src> {
 		let mut fields = Vec::new();
 
-		for field in self.fields {
-			fields.push((field.0.into_config(), field.1.into_config(state)));
+		for field in ty.fields.iter() {
+			fields.push((field.0.name, self.ty(&field.1)));
 		}
 
 		Struct { fields }
 	}
-}
 
-impl<'src> SyntaxRange<'src> {
-	fn into_config(self, state: &mut ConvertState<'src>) -> Range {
-		let range = match self.kind {
-			SyntaxRangeKind::None => Range::new(None, None),
-			SyntaxRangeKind::Exact(num) => Range::new(Some(num.into_config()), Some(num.into_config())),
-			SyntaxRangeKind::WithMin(min) => Range::new(Some(min.into_config()), None),
-			SyntaxRangeKind::WithMax(max) => Range::new(None, Some(max.into_config())),
-			SyntaxRangeKind::WithMinMax(min, max) => Range::new(Some(min.into_config()), Some(max.into_config())),
-		};
+	fn ty_has_unbounded_ref(
+		&self,
+		name: &'src str,
+		ty: &SyntaxTy<'src>,
+		searched: &mut HashSet<&'src str>,
+	) -> Option<SyntaxIdentifier<'src>> {
+		match &ty.kind {
+			SyntaxTyKind::Arr(ty, len) => {
+				let len = len.map(|len| self.range(&len)).unwrap_or_default();
+
+				// if array does not have a min size of 0, it is unbounded
+				if len.min().is_some_and(|min| min != 0.0) {
+					self.ty_has_unbounded_ref(name, ty, searched)
+				} else {
+					None
+				}
+			}
+
+			SyntaxTyKind::Ref(ref_ty) => {
+				let ref_name = ref_ty.name;
+
+				match ref_name {
+					ref_name if ref_name == name => Some(*ref_ty),
+
+					"boolean" | "Color3" | "Vector3" | "AlignedCFrame" | "CFrame" | "unknown" => None,
```

Comment:
Should we put these somewhere else so we don't have to hardcode primatives in two places?

Reply:
I'd like that, but idk what to do for it

Reply:
This is a terrible idea please don't